vue-app/src/components/WalletModal.vue

```diff
@@ -0,0 +1,218 @@
+<template>
+  <div>
+    <div v-if="connectingWallet" class="modal-body loading">
+      <loader />
+      <p>Connecting wallet</p>
+    </div>
+    <div v-else class="modal-body">
+      <div class="top">
+        <p>Connect to a wallet</p>
+        <img class="pointer" src="@/assets/close.svg" @click="$emit('close')" />
+      </div>
+      <div v-if="windowEthereum" class="option" @click="connectMetaMask()">
+        <p>MetaMask</p>
+        <img height="24px" width="24px" src="@/assets/metamask.png" />
+      </div>
+      <div v-else class="option" @click="redirectToMetamaskWebsite()">
+        <p>Install MetaMask</p>
+        <img height="24px" width="24px" src="@/assets/metamask.png" />
+      </div>
+      <div class="option" @click="connectWalletConnect()">
+        <p>WalletConnect</p>
+        <img height="24px" width="24px" src="@/assets/walletConnectIcon.svg" />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Web3Provider } from '@ethersproject/providers'
+import Vue from 'vue'
+import Component from 'vue-class-component'
+import { Prop } from 'vue-property-decorator'
+
+// API
+import { providers } from '@/api/core'
+import { LOGIN_MESSAGE, User, getProfileImageUrl } from '@/api/user'
+
+// Components
+import Loader from '@/components/Loader.vue'
+
+// Store
+import {
+  LOAD_USER_INFO,
+  LOAD_CART,
+  LOAD_COMMITTED_CART,
+  LOAD_CONTRIBUTOR_DATA,
+  LOGIN_USER,
+} from '@/store/action-types'
+import { SET_CURRENT_USER } from '@/store/mutation-types'
+
+// Utils
+import { sha256 } from '@/utils/crypto'
+
+@Component({
+  components: {
+    Loader,
+  },
+})
+export default class WalletModal extends Vue {
+  @Prop() updateProfileImage!: (profileImageUrl: string | null) => void
+
+  connectingWallet = false
+
+  get windowEthereum(): any {
+    return (window as any).ethereum
+  }
+
+  async connectMetaMask() {
```

Comment:
Just a refactor suggestion, we could modify a bit `connectMetaMask` and `connectWalletConnect` to follow the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle and make it easier in the future to add a new wallet/provider.

```ts
async connectMetaMask() {
    if (!this.windowEthereum || !this.windowEthereum.request) {
      return
    }

    this.connectingWallet = true
    let walletAddress
    try {
      ;[walletAddress] = await this.windowEthereum.request({
        method: 'eth_requestAccounts',
      })
    } catch (error) {
      // Access denied
      return
    }

    const user = await this.connectAccount(this.windowEthereum, walletAddress)

    if (user) {
      await this.loginUser(user)
    }
  }

  async connectWalletConnect() {
    this.connectingWallet = true
    await providers.walletconnectProvider.enable()

    const user = await this.connectAccount(
      providers.walletconnectProvider,
      providers.walletconnectProvider.accounts[0]
    )

    if (user) {
      await this.loginUser(user)
    }
  }

  async connectAccount(
    provider: any,
    account: string
  ): Promise<User | undefined> {
    let signature
    try {
      signature = await provider.request({
        method: 'personal_sign',
        params: [LOGIN_MESSAGE, account],
      })
    } catch (error) {
      // Signature request rejected
      this.connectingWallet = false
      return
    }

    const user: User = {
      walletProvider: new Web3Provider(provider),
      walletAddress: account,
      encryptionKey: sha256(signature),
      isVerified: null,
      balance: null,
      contribution: null,
    }

    return user
  }

  async loginUser(user: User): Promise<void> {
    this.$emit('connected')
    this.connectingWallet = false

    const url = await getProfileImageUrl(user.walletAddress)
    this.updateProfileImage(url)

    this.$store.commit(SET_CURRENT_USER, user)
    await this.$store.dispatch(LOGIN_USER)
    if (this.$store.state.currentRound) {
      // Load cart & contributor data for current round
      this.$store.dispatch(LOAD_USER_INFO)
      this.$store.dispatch(LOAD_CART)
      this.$store.dispatch(LOAD_COMMITTED_CART)
      this.$store.dispatch(LOAD_CONTRIBUTOR_DATA)
    }
    this.$emit('close')
  }
```

I haven't tested it but I wanted to share it with you.

Reply:
Not implemented. Took a diff approach using a plugin.